sandman/source/shell/input_window_eventful_buffer.h

```diff
@@ -30,23 +31,68 @@ class Shell::InputWindow::EventfulBuffer
 		using OnClearListener = void (*)();
 		using OnDecrementStringLengthListener = void (*)(typename Data::size_type const newStringLength);
 
+		template <typename EventListenerT>
```

Comment:
Personally I would just go the simpler route of checking for nullptr at call time.